### PR TITLE
added sign-in edge case tests

### PR DESCRIPTION
**Pull request by nasirkhanbeis**
Sign Tests - edge cases

## Description
Added further tests around validating error messages

## Checklist:

- [ ] Automated checks are passing locally.
- [ ] CHANGELOG updated if change is worth telling users about.
### General testing
- [ ] Test without javascript
- [ ] Test on small screen
### Accessibility testing
- [ ] Works keyboard only
- [ ] Tested with one screen reader
- [ ] Zoom page to 400% - content still visible
- [ ] Disable css - does content make sense and appear in a logical order?
- [ ] Passes automated checker (automated in build or manual)

### CODE REVIEW

File: psd-web/spec/features/sign_in_spec.rb [hunk 1]
```diff
@@ -42,5 +42,39 @@
...
+        visit root_path
+        click_on "Sign in to your account"
+        fill_in "Email address", with: " "
+        fill_in "Password", with: " "
+        click_on "Continue"
+        expect(page).to have_css(".govuk-error-message", text: "Enter your email address")
+        expect(page).to have_css(".govuk-error-message", text: "Enter your password")
+      end
+    end
+    context "when password field is left empty" do
```
Comment (StupidCodeFactory) [inline psd-web/spec/features/sign_in_spec.rb]:
```suggestion

    context "when password field is left empty" do
```

File: psd-web/spec/features/sign_in_spec.rb [hunk 2]
```diff
@@ -42,5 +42,39 @@
...
+        click_on "Continue"
+        expect(page).to have_css(".govuk-error-summary__list", text: "Enter your email address in the correct format, like name@example.com")
+        expect(page).to have_css(".govuk-error-message", text: "Enter your email address in the correct format, like name@example.com")
+      end
+    end
+
+    context "when email and password fields left empty" do
+      scenario "shows error messages" do
+        visit root_path
+        click_on "Sign in to your account"
```
Comment (StupidCodeFactory) [inline psd-web/spec/features/sign_in_spec.rb]:
```suggestion

        click_on "Sign in to your account"
```

File: psd-web/spec/features/sign_in_spec.rb [hunk 3]
```diff
@@ -42,5 +42,39 @@
...
+        fill_in "Email address", with: " "
+        fill_in "Password", with: " "
+        click_on "Continue"
+        expect(page).to have_css(".govuk-error-message", text: "Enter your email address")
```
Comment (StupidCodeFactory) [inline psd-web/spec/features/sign_in_spec.rb]:
```suggestion

        expect(page).to have_css(".govuk-error-message", text: "Enter your email address")
```

File: psd-web/spec/features/sign_in_spec.rb [hunk 4]
```diff
@@ -42,5 +42,39 @@
...
+        expect(page).to have_css(".govuk-error-message", text: "Enter your password")
+      end
+    end
+    context "when password field is left empty" do
+      scenario "shows an error messages" do
+        visit root_path
+        click_on "Sign in to your account"
+        fill_in "Email address", with: user.email
+        fill_in "Password", with: " "
+        click_on "Continue"
```
Comment (StupidCodeFactory) [inline psd-web/spec/features/sign_in_spec.rb]:
```suggestion
        click_on "Continue"
        
```

File: psd-web/spec/features/sign_in_spec.rb [hunk 5]
```diff
@@ -42,5 +42,39 @@
...
+        expect(page).to have_css(".govuk-error-message", text: "Enter your email address")
+        expect(page).to have_css(".govuk-error-message", text: "Enter your password")
+      end
+    end
+    context "when password field is left empty" do
+      scenario "shows an error messages" do
+        visit root_path
+        click_on "Sign in to your account"
```
Comment (StupidCodeFactory) [inline psd-web/spec/features/sign_in_spec.rb]:
```suggestion
        click_on "Sign in to your account"
        
```

File: psd-web/spec/features/sign_in_spec.rb [hunk 6]
```diff
@@ -42,5 +42,39 @@
         expect(page).to have_css("span#password-error", text: "")
       end
     end
+    context "when email address is not in correct format" do
+      scenario "shows an error message" do
+        visit root_path
+        click_on "Sign in to your account"
+        fill_in "Email address", with: "test.email"
+        fill_in "Password", with: "password"
+        click_on "Continue"
```
Comment (StupidCodeFactory) [inline psd-web/spec/features/sign_in_spec.rb]:
```suggestion
        click_on "Continue"
        
```

File: psd-web/spec/features/sign_in_spec.rb [hunk 7]
```diff
@@ -42,5 +42,39 @@
         expect(page).to have_css("span#password-error", text: "")
       end
     end
+    context "when email address is not in correct format" do
+      scenario "shows an error message" do
+        visit root_path
+        click_on "Sign in to your account"
```
Comment (StupidCodeFactory) [inline psd-web/spec/features/sign_in_spec.rb]:
```suggestion
        click_on "Sign in to your account"
        
```

**Comment by nasirkhanbeis** [comment]
@slorek another small pr, appreciate if you want take a quick look before I want to merge this.
